core/store/src/config.rs

```diff
@@ -129,13 +135,12 @@ pub struct StateSnapshotConfig {
 
 #[derive(Clone, Debug, Default, serde::Serialize, serde::Deserialize)]
 pub enum StateSnapshotType {
-    /// Consider this as the default "disabled" option. We need to have snapshotting enabled for resharding
-    /// State snapshots involve filesystem operations and costly IO operations.
-    ForReshardingOnly,
     /// This is the "enabled" option where we create a snapshot at the beginning of every epoch.
-    /// Needed if a node wants to be able to respond to state part requests.
     #[default]
-    EveryEpoch,
+    #[serde(alias = "EveryEpoch")]
+    Enabled,
+    #[serde(alias = "ForReshardingOnly")]
```

Comment:
nit: add a TODO to say we can remove this starting from.. 2.9 release? (this change will be likely shipped in 2.7)